test/functions/vass/type-and-param-function-arg-conversions.chpl

```diff
@@ -15,12 +15,12 @@ proc typeFunUse(i:int)    type  return [1..i] int;
 
 var sv1, sv2, sv3: sync int;
 
-writeln(paramFun(sv1));
+writeln(paramFun(sv1.readXX()));
```

Comment:
@bradcray - I see that `readXX` is necessary here (vs `readFE`) but it seems to me that adding it makes the test no longer check what it was intended too. The thing is, I'm not sure how to do the check it wants to without user-defined implicit conversions or using the deprecated sync operations.

Reply:
Huh, interesting...  Is it clear how/why this worked previously, with the compiler-injected .readFE()?  Is it just that we resolve the param function prior to inserting the .readFE() call?  (but then why would the type be interpreted as `int` rather than `sync int`).

Let's see what @vasslitvinov would like to do here.  In some respects, I consider this a positive outcome in that it seems to make `param` and `type` functions more symmetrical in this respect (where I read the original test as saying "Is this seeming asymmetry weird?  Should we change something?").  Left to my own devices, I'd make the param version more symmetrical with the type cases by filling the variables prior to the call and using .readFE() (but apparently didn't spend enough time thinking about this to do it previously).

I love the description on the PR that merged it:  https://github.com/chapel-lang/chapel/pull/249  :D